### PR TITLE
Support bearer auth in python client

### PR DESCRIPTION
**Pull request by dimas-b**
* Also adjust outdated Basic and AWS auth code

Related to #1768 

---
This change is [Reviewable](https://reviewable.io/reviews/projectnessie/nessie/1863)

### CODE REVIEW

File: python/pynessie/_endpoints.py [hunk 1]
```diff
@@ -8,8 +8,10 @@
 
 import requests
 import simplejson as jsonlib
+from confuse import Configuration
```
Comment (rymurr) [inline python/pynessie/_endpoints.py:11]: this was a set of functions to allow for composing functions w/o a class? Why make a class now? This class is now nearly identical to `NessieClient` so we either refactor into that class or keep these as functions

File: python/pynessie/nessie_client.py [hunk 2]
```diff
@@ -51,21 +38,24 @@ class NessieClient(object):
...
-        self._ssl_verify = config["verify"].get(bool)
+        self._endpoints = EndPoints(config)
         self._commit_id: str = cast(str, None)
 
         try:
             self._base_branch = config["default_branch"].get()
         except confuse.exceptions.NotFoundError:
             self._base_branch = None
 
+    def base_url(self: "NessieClient") -> str:
```
Comment (rymurr) [inline python/pynessie/nessie_client.py:49]: if we want to expose this as a method I would prefer as a `property`

File: python/setup.cfg [hunk 3]
```diff
@@ -28,6 +28,3 @@ per-file-ignores = tests/*:S101
 
 [aliases]
 test = pytest
-
-[tool:pytest]
-collect_ignore = ['setup.py']
```
Comment (rymurr) [inline python/setup.cfg:33]: why remove this?

**Review by rymurr — CHANGES_REQUESTED** [review]

File: python/pynessie/auth/aws.py [hunk 4]
```diff
@@ -1,12 +0,0 @@
-# -*- coding: utf-8 -*-
```
[the remark this answers is not in the document]
Comment (dimas-b) [reply python/pynessie/auth/aws.py:1]: It is not used currently and was not called before. Shall I re-enable AWS auth?

File: python/pynessie/auth/basic.py [hunk 5]
```diff
@@ -1,20 +0,0 @@
-# -*- coding: utf-8 -*-
```
[the remark this answers is not in the document]
Comment (dimas-b) [reply python/pynessie/auth/basic.py:1]: Basic auth is not supported in production. The only potential use case now is to run the Nessie server in `dev` mode and explicitly enable basic auth in Quarkus config... I assumed it was an edge case.

File: python/setup.cfg (hunk 3 above)
Comment (dimas-b) [reply python/setup.cfg:33]: Test tools give warnings that this setting is present, but is not necessary

Comment (dimas-b) [reply python/setup.cfg:33]:
```
========================================================== warnings summary ===========================================================
../../../.pyenv/versions/3.9.6/envs/nessie@3.9.6-dev/lib/python3.9/site-packages/_pytest/config/__init__.py:1233
  /home/dmitri/.pyenv/versions/3.9.6/envs/nessie@3.9.6-dev/lib/python3.9/site-packages/_pytest/config/__init__.py:1233: PytestConfigWarning: Unknown config option: collect_ignore
  
    self._warn_or_fail_if_strict(f"Unknown config option: {key}\n")

-- Docs: https://docs.pytest.org/en/stable/warnings.html
```

File: python/pynessie/_endpoints.py (hunk 1 above)
Comment (dimas-b) [reply python/pynessie/_endpoints.py]: I went back to individual function in `_endpoints` in cce999859b660e462b538b51b451f438e210ee6e (mainly to reduce the amount of changes in the PR overall).

File: python/pynessie/nessie_client.py (hunk 2 above)
Comment (dimas-b) [reply python/pynessie/nessie_client.py]: I rolled back this changes in cce999859b660e462b538b51b451f438e210ee6e

File: python/pynessie/auth/aws.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (rymurr) [reply python/pynessie/auth/aws.py:1]: yeah I think so, it exists in java and was being called in `auth.__init__`

File: python/pynessie/auth/basic.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (rymurr) [reply python/pynessie/auth/basic.py:1]: i think it likely is, but its hard to ship a client library and easy to enable it in the server. Maybe best to ship it, don't feel too strongly though

File: python/setup.cfg (hunk 3 above)
Comment (rymurr) [reply python/setup.cfg:33]: interesting, must have been deprecated in a version bump :+1:

File: python/pynessie/auth/basic.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (dimas-b) [reply python/pynessie/auth/basic.py:1]: Easy to re-add, will do so.

File: python/pynessie/auth/aws.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (dimas-b) [reply python/pynessie/auth/aws.py]: AWS auth restored in 84bff00fe2437e3fed49020afe2141760962ce97

File: python/pynessie/auth/basic.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (dimas-b) [reply python/pynessie/auth/basic.py:1]: Basic auth restored in 804c7f9612d0c4fb0b8c9905a0169c45b5f35956

**Comment by dimas-b** [comment]
Squashed and rebased

**Review by rymurr — APPROVED** [review]
